Sources/SwiftProcessing/Core/UIKitElements/Capture.swift

```diff
@@ -0,0 +1,310 @@
+                self.showDisabledCameraAlert(completion: completion)
+            }
+        }
+    }
+    
+    private func configureCaptureSession() {
+        guard let previewView = previewView else { return }
+        // Define the capture device we want to use
+        
+        guard let camera = AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: cameraPosition) else {
+            let error = NSError(domain: "", code: 0, userInfo: [NSLocalizedDescriptionKey : "No front camera available"])
+            show(error: error)
+            return
+        }
+        
+        // Connect the camera to the capture session input
+        do {
+            
+            try camera.lockForConfiguration()
+            defer { camera.unlockForConfiguration() }
+            
+            if camera.isFocusModeSupported(.continuousAutoFocus) {
+                camera.focusMode = .continuousAutoFocus
+            }
+            
+            if camera.isExposureModeSupported(.continuousAutoExposure) {
+                camera.exposureMode = .continuousAutoExposure
+            }
+            
+            let cameraInput = try AVCaptureDeviceInput(device: camera)
+            session.addInput(cameraInput)
+            
+        } catch {
+            show(error: error as NSError)
+            return
+        }
+        
+        // Create the video data output
+        let videoOutput = AVCaptureVideoDataOutput()
+        videoOutput.setSampleBufferDelegate(self, queue: dataOutputQueue)
+        videoOutput.videoSettings = [kCVPixelBufferPixelFormatTypeKey as String: kCVPixelFormatType_32BGRA]
+        
+        // Add the video output to the capture session
+        session.addOutput(videoOutput)
+        
+        let videoConnection = videoOutput.connection(with: .video)
+        videoConnection?.videoOrientation = .portrait
+        
+        // Configure the preview layer
+        let previewLayer = AVCaptureVideoPreviewLayer(session: session)
+        previewLayer.videoGravity = .resizeAspectFill
+        previewLayer.frame = previewView.bounds
+        previewView.layer.insertSublayer(previewLayer, at: 0)
+        self.previewLayer = previewLayer
+    }
+}
+
+extension Camera: AVCaptureVideoDataOutputSampleBufferDelegate {
+    public func captureOutput(_ output: AVCaptureOutput, didOutput sampleBuffer: CMSampleBuffer, from connection: AVCaptureConnection) {
+        guard   captureCompletionBlock != nil,
+            let outputImage = UIImage(sampleBuffer: sampleBuffer, orientation: snapshotImageOrientation) else { return }
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            if let captureCompletionBlock = self.captureCompletionBlock{
+                captureCompletionBlock(outputImage)
+            }
+            self.captureCompletionBlock = nil
+        }
+    }
+}
+
+// Navigation
+
+extension Camera {
+    
+    private func show(alert: UIAlertController) {
```

Comment:
the show functions + topViewController extension would make sense in another file named Alerts. I can see them being useful for other elements in the future.

In Camera, you could instead create an instance of your Alerts object and leverage from the functions from there.